src/open-uri.c

```diff
@@ -533,11 +534,9 @@ find_recommended_choices (const char *scheme,
   result = g_new (char *, n_choices + 1);
   for (l = infos, i = 0; l; l = l->next)
     {
-      info = l->data;
-      result[i++] = get_app_id (info);
+      result[i++] = get_app_id ((GAppInfo *)l->data);
     }
```

Comment:
Since you're cleaning up this loop, I think it can be simplified to the following:

```c
for (l = infos, i = 0; l; l = l->next)
  result[i++] = get_app_id (G_APP_INFO (l->data));
```